client/catapult/tests/catapult/utils/CatapultExceptionTests.cpp

```diff
@@ -97,7 +98,15 @@ namespace catapult {
 			if (std::string::npos != foundTraits) {
 				auto traitsEnd = functionFullName.find("]", foundTraits);
 				auto functionTraits = functionFullName.substr(foundTraits + 2, traitsEnd - foundTraits - 2);
-				oss << "<" << functionTraits << ">";
+				oss << "<";
+
+	// A bug in gcc 12 where the trait information is missing the namespace.
+	// Gcc 12 returns this - [with TTraits = {anonymous}::RuntimeErrorTraits]
+	#if (__GNUC__ == 12)
```

Comment:
if this is GCC 12 bug, shouldn't it be `#if 12 != __GNUC_`

Reply:
The issue is that the namespace information is in the exception message but missing from the boost function.
if GCC 12 then add the missing namespace to the trait so it will match the exception info

Reply:
so, you mean it is a bug in boost?

Reply:
maybe. idk.  I thought compiler since it works in gcc11.